mono/metadata/loader.c

```diff
@@ -1414,13 +1479,55 @@ pinvoke_probe_for_module (MonoImage *image, const char*new_scope, const char *im
 			break;
 #endif
 		}
-		
+		changed = TRUE;
 		if (is_absolute) {
 			if (!dir_name)
 				dir_name = g_path_get_dirname (file_name);
 			if (!base_name)
 				base_name = g_path_get_basename (file_name);
 		}
+	} while (0);
```

Comment:
There's a complicated bit here in an attempt to minimize line damage.
```
  do { ... 
    switch { cases ... } ;
    ... success code ...
  } while (0);
```

This is because the old code used `break` and `continue`.  `continue` means - nothing changed in this phase, so try a different name transformation.  `break` means - I modified the path in some way and we should probe there.

So since we don't have a `for` loop in this function anymore, we do a `do { ... } while (0)`.  The `break` breaks out of the `switch` and sets `changed = TRUE`, whereas the `continue` tries to run another loop iteration (which doesn't work because `(0)`) and bypasses setting `changed = TRUE`.  So after the loop `changed == TRUE` only in the cases where we hit a `break`, and not in the cases where we hit a `continue`.

Obviously we can make this cleaner, but at the cost of line damage.  Maybe worth it?

Reply:
There is already a lot of line damage, so just for go it.

Reply:
ok. made it use a normal switch.